src/main/java/io/cdap/plugin/ariba/source/AribaStructuredTransformer.java

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.ariba.source;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.api.common.Bytes;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.plugin.ariba.source.util.ResourceConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * This {@code AribaStructuredTransformer} contains the logic to
+ * convert Ariba record to {@code StructuredRecord}
+ */
+public class AribaStructuredTransformer {
+  private static final Logger LOG = LoggerFactory.getLogger(AribaStructuredTransformer.class);
+
+  public StructuredRecord readFields(JsonNode nativeRawRecord, Schema outputSchema) {
+    StructuredRecord.Builder recBuilder = StructuredRecord.builder(outputSchema);
+    List<Schema.Field> cdapFields = outputSchema.getFields();
+
+    cdapFields.forEach(field -> {
+      String fieldName = field.getName();
+      Object fieldValStr = getFieldNativeValue(nativeRawRecord, fieldName);
```

Comment:
why is it named fieldValStr if the type is not necessarily string?

Reply:
Fixed